config/boards/olimex-teres-a64.conf

```diff
@@ -0,0 +1,36 @@
+# OLIMEX Teres-A64
+# Maintainer: Jacob "Kreyren" Hrbek <kreyren@fsfe.org>
+
+# Optimized for Mainboard Rev.C
+
+BOARD_NAME="OLIMEX Teres A64" # Allwinner A64 quad core 2GB SoC Wi-Fi/BT
+
+# Specified in https://linux-sunxi.org/Allwinner_SoC_Family
+BOARDFAMILY="sun50iw1"
+
+# Has custom default configuration of the u-boot that should be used -- https://linux-sunxi.org/Olimex_Teres-A64#Mainline_U-Boot
+BOOTCONFIG="teres_i_defconfig"
+
+# Observed issues on insufficient size, thus enforced -- https://github.com/OLIMEX/DIY-LAPTOP/issues/52
+BOOTSIZE="256"
+
+# Seems to be the most reliable and bootloader should stay as separate partition
```

Comment:
BS. Tens of thousands of Allwinner devices are running perfectly fine with one ext4 partition.

Reply:
BOOTSIZE: The issue is when using debian/ubuntu's software upgrader as it seems to be exponentially demanding more space with each iteration so upgrading from xenial to jammy results in broken system without at least 250MB (see linked issue)

Separate FAT partition is what is used in the official OLIMEX image and i want the user to have the option to do rootfs encryption painlessly

Reply:
No reply for 4 days, considering stale

Reply:
> The issue is when using debian/ubuntu's software upgrader as it seems to be exponentially

This is expected and understandable, but if we use ext4 boot partition, this paramater is not needed. Check Pinebook A64 which are very close ...

> is what is used in the official 

(i only scanned this text, but its enough to generate context)
https://www.cprime.com/resources/blog/hardware-vs-software-development-similarities-and-differences/

Armbian people are software development experts in hardware / embedded Linux and Olimex people are hardware development experts. We work together, but have different focus, different goals, ... Their official software is not a reference for us.

> and i want the user to have the option to do rootfs encryption painlessly

This is not making it easier. If you choose rootfs encryption or btrfs, boot partition is created in the process automatically. Also when installing to different media.

Reply:
> This is expected and understandable, but if we use ext4 boot partition, this paramater is not needed. Check Pinebook A64 which are very close ... -- @igorpecovnik (https://github.com/armbian/build/pull/4807#discussion_r1103794744)

I am aware, but that doesn't address the issue of Full Disk Encryption as in such scenarios you want the boot partition separate so that it can stay unencrypted to avoid inputing the password multiple times.

> Armbian people are software development experts in hardware / embedded Linux and Olimex people are hardware development experts. We work together, but have different focus, different goals, ... Their official software is not a reference for us. -- @igorpecovnik (https://github.com/armbian/build/pull/4807#discussion_r1103794744)

Disagree for the rationale above.

> This is not making it easier. If you choose rootfs encryption or btrfs, boot partition is created in the process automatically. Also when installing to different media. -- @igorpecovnik (https://github.com/armbian/build/pull/4807#discussion_r1103794744)

I care about post-installation experience as the solution with non-separate /boot partition would require the user to restructure the partitions which is not user-friendly and painful to do.

Reply:
Btw. providing a half-read article and self-proclaiming to be an expert on a subject you clearly don't understand judging by the disfunctional armbian's Full Disk Encryption backend is why computer science is affected by [replication crisis](https://en.wikipedia.org/wiki/Replication_crisis).

Consider adjusting your approach and how you present yourself in this industry.

Reply:
> Disagree for the rationale above.

A few people can't cover the work of 1000 people. A common misperception from average Linux persons we are seeing daily. A man complains: "this is not working ...", which is normal. You have to tell this to someone. In open source its always a big pile of things that doesn't work and some never will ... then he comes back in few days "is it fixed?" ... It is totally useless explaining that this and that problem requires months of work with resources we don't have, while there are thousands of already known problems (users don't understand what is our work and what not) and we are working over what is possible ... Which we will not even started as - with no income for making public code - we can't afford to endlessly support end users addition & hobby. Line is set with a [licence](https://github.com/armbian/build/blob/main/LICENSE). 

 Are you a Linux newbie? Then its forgiven as you didn't know. If not, then stop with BS. I am tired of explaining this over and over again.

> not meant to be aggressive

Keep in mind that demanding attention and support is aggressive act in this world.

All we expect from you is to fix what was requested or close the PR.

> disfunctional armbian's Full Disk Encryption

Armbian technical documentation, for which you have absolutely no rights to complain about, is saying: 

CRYPTROOT_ENABLE ( yes | no ): enable LUKS encrypted rootfs
Warning: This feature was added as community contribution and mostly functional. Under some circumstances though the prompt will not be shown. Therefore it should be **considered experimental**.

If we had someone dealing with this, but we don't, it would be supported & functional, label "experimental" removed ...

Reply:
> Bla bla bla you are stupid krey, additional tantrum

Ignored as it's not worth my time and additional hassle i am here to merge not to argue about bullshit

---

I have branch that 'almost' fixes the encryption to be reliable where the only issue is that docker doesn't provide /dev/loopXpY once it was created that i am in contact with docker upstream to address, the state is that changed base works if that was addressed, but looks like trash so once resolved i will refactor it and then submit.

Without separate bootloader the encryption will ask for password multiple times which is really annoying + It makes it more comfortable for the user to fix issues and use armbian as a donor for other distros such as the archlinux/parabola support i used this for yesterday.